app/components/primer/beta/blankslate.rb

```diff
@@ -0,0 +1,241 @@
+
+          Primer::OcticonComponent.new(**system_arguments)
+        },
+        spinner: lambda { |**system_arguments|
+          system_arguments[:mb] = 3
+
+          Primer::SpinnerComponent.new(**system_arguments)
+        },
+        image: lambda { |**system_arguments|
+          system_arguments[:mb] = 3
+          system_arguments[:size] = "56x56"
+
+          Primer::Image.new(**system_arguments)
+        }
+      }
+
+      # Required Title.
+      #
+      # @param tag [String]  <%= one_of(Primer::HeadingComponent::TAG_OPTIONS) %>
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :title, lambda { |tag:, **system_arguments|
+        system_arguments[:tag] = tag
+        system_arguments[:mb] = 1
+        system_arguments[:classes] = class_names("h2", system_arguments[:classes])
+
+        Primer::HeadingComponent.new(**system_arguments)
+      }
+
+      # Optional Description.
+      #
+      # - The description should always be informative and actionable.
+      # - Don't use phrases like "You can".
+      #
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :description, lambda { |**system_arguments|
+        system_arguments[:tag] = :p
+
+        Primer::BaseComponent.new(**system_arguments)
+      }
+
+      # Optional Primary action
+      #
+      # Use this slot to set a call to action for users.
+      #
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :primary_action, lambda { |href:, **system_arguments|
+        system_arguments[:tag] = :a
+        system_arguments[:href] = href
+        system_arguments[:my] = 3
+        system_arguments[:variant] = :large
+        system_arguments[:scheme] ||= :primary
+
+        Primer::ButtonComponent.new(**system_arguments)
+      }
+
+      # Optional Secondary action
+      #
+      # Use this slot to provide more information for the user.
+      #
+      # @param href [String] URL to be used for the link.
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :secondary_action, lambda { |href:, **system_arguments|
+        system_arguments[:href] = href
+        # Padding is needed to increase touch area.
+        system_arguments[:p] = 3
+
+        Primer::LinkComponent.new(**system_arguments)
+      }
+
+      #
+      # @example Basic
+      #   <%= render Primer::Beta::Blankslate.new do |c| %>
+      #     <% c.title(tag: :h2).with_content("Title") %>
+      #     <% c.description { "Description"} %>
+      #   <% end %>
+      #
+      # @example Icon
+      #   @description
+      #     Add an `icon` to give additional context. Refer to the [Octicons](https://primer.style/octicons/) documentation to choose an icon.
+      #   @code
+      #     <%= render Primer::Beta::Blankslate.new(icon: :globe) do |c| %>
+      #       <% c.graphic_icon(icon: :globe) %>
```

Comment:
Is this example correct? I'd expect us to pass `icon` to the slot, not the component itself.